src/includes/proc-service.h

```diff
@@ -14,6 +18,41 @@
 
 class process_service;
 
+#if SUPPORT_CAPABILITIES
+// A thin wrapper around the structure to avoid having to worry about freeing
+// it in various places (move semantics and destructor will take care of it)
+struct cap_iab_wrapper {
+    cap_iab_wrapper() {}
+    cap_iab_wrapper(std::string const &str) noexcept {
+        if (str.empty()) return;
+        iab = cap_iab_from_text(str.c_str());
+    }
+
+    cap_iab_wrapper(cap_iab_wrapper const &) = delete;
+    cap_iab_wrapper(cap_iab_wrapper &&v) noexcept: iab(v.iab) {
+        v.iab = nullptr;
+    }
+
+    cap_iab_wrapper &operator=(cap_iab_wrapper const &) = delete;
+    cap_iab_wrapper &operator=(cap_iab_wrapper &&v) noexcept {
+        iab = v.iab;
+        v.iab = nullptr;
+        return *this;
+    }
+
+    ~cap_iab_wrapper() noexcept {
+        cap_free(iab);
```

Comment:
It's not clear from libcap documentation (unless I've missed something) that it is valid to call `cap_free` with a null argument, so I think it is better to guard against that case.

Reply:
the implementation explicitly checks it, since 1.95 release in 2007, but sure